realtime/messages.go

```diff
@@ -158,16 +158,16 @@ func (c *Client) UnPinMessage(message *models.Message) error {
 // Returns a buffered channel
 //
 // https://rocket.chat/docs/developer-guides/realtime-api/subscriptions/stream-room-messages/
-func (c *Client) SubscribeToMessageStream(channel *models.Channel, msgChannel chan models.Message) error {
+func (c *Client) SubscribeToMessageStream(channel *models.Channel) (chan models.Message, error) {
```

Comment:
Being able to pass the channel to use in is actually one of the benefits here.  It would let you set up a loop to handle all messages.  

Kinda like the example I sent in #golangsdk

```
```
ch := make(chan models.Message)
for _, channel := range channels {
	_ = client.SubscribeToMessageStream(&channel, ch)

	fmt.Println("Started listening channel: " + channel.ID)
}

go func () {
      for msg := range ch {
			fmt.Printf("Message [%s]: %+v\n", msg.RoomID, msg)
      }
}()
```

Here that loop handles all messages from all channels.  Of course you could choose to divide it up and handle different rooms separately if you wished.

Reply:
I'd personally be fine with this either way, because it's not great task to merge channels anyway, it's just that returning channels seemed ergonomically better and also matched function documentation.

If the other change (to the extractor part) makes sense to you I can resubmit these changes without returning channel.

Reply:
Yeah I think the better thing here is to keep passing it in so we don't break existing that are using it this way